mysql/datadog_checks/mysql/mysql.py

```diff
@@ -124,6 +127,9 @@ def check(self, _):
             finally:
                 self._conn = None
 
+    def cancel(self):
+        self._statement_samples.cancel()
```

Comment:
@olivielpeau @hush-hush I added support for `cancel` here. I think we should still keep the "stop thread if it notices the check is no longer running" logic as a fallback if the main check stops running for any other reason (agent overload, or a bug somewhere where the check gets stuck).

Reply:
ok, LGTM.

I don't have a strong opinion about the fallback. Just a couple of thoughts: if the check is stuck it'll cause other problems (it'll hog a check runner, etc), but the fallback would at least free up some resources that's true. If the check is inactive because of an agent overload, and the fallback mechanism stops the background thread, you'll want to make sure the background thread can be started up again cleanly when the check is run again.

Reply:
> you'll want to make sure the background thread can be started up again cleanly when the check is run again.

Yes that is the current behavior. On every check run the check will start up the thread if it is not currently running.